packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,694 @@
+#include "DetectorUtils.h"
+#include <algorithm>
+#include <limits>
+#include <numeric>
+#include <rnexecutorch/data_processing/Numerical.h>
+#include <rnexecutorch/models/ocr/Constants.h>
+#include <rnexecutorch/models/ocr/Types.h>
+#include <unordered_set>
+namespace rnexecutorch::ocr {
+
+std::vector<cv::Point2f>
+cvPointsFromPoints(const std::array<Point, 4> &points) {
+  std::vector<cv::Point2f> cvPoints;
+  for (const Point &point : points) {
+    cvPoints.emplace_back(point.x, point.y);
+  }
+  return cvPoints;
+}
+
+std::array<Point, 4> pointsFromCvPoints(cv::Point2f cvPoints[4]) {
+  std::array<Point, 4> points;
+#pragma unroll
+  for (std::size_t i = 0; i < points.size(); ++i) {
+    points[i] = {.x = cvPoints[i].x, .y = cvPoints[i].y};
+  }
+  return points;
+}
+
+std::pair<cv::Mat, cv::Mat> interleavedArrayToMats(std::span<const float> data,
+                                                   cv::Size size) {
+  cv::Mat mat1 = cv::Mat(size.height, size.width, CV_32F);
+  cv::Mat mat2 = cv::Mat(size.height, size.width, CV_32F);
+
+  for (std::size_t i = 0; i < data.size(); i++) {
+    const float value = data[i];
+    const int x = (i / 2) % size.width;
+    const int y = (i / 2) / size.width;
+
+    if (i % 2 == 0) {
+      mat1.at<float>(y, x) = value;
+    } else {
+      mat2.at<float>(y, x) = value;
+    }
+  }
+  return {mat1, mat2};
+}
+
+// Create a segmentation map for the current component.
+// Background is 0, (black), foreground is 255 (white)
+cv::Mat createSegmentMap(cv::Mat &mask, cv::Size mapSize,
```

Comment:
I guess that this should work
```suggestion
cv::Mat createSegmentMap(const cv::Mat &mask, cv::Size mapSize,
```